hq/test/logic/SnykDisplayTest.scala

```diff
@@ -0,0 +1,291 @@
+package logic
+
+import model._
+import org.scalatest.{FreeSpec, Matchers}
+import utils.attempt.AttemptValues
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class SnykDisplayTest extends FreeSpec with Matchers with AttemptValues {
+
+  "find organisation" in {
+    val organisation = SnykDisplay.getOrganisation(SnykDisplayTest.mockGoodOrganisationResponse, SnykOrganisationName("guardian"))
+    organisation.value shouldBe SnykOrganisation("guardian", "1111111111")
+  }
+
+  "fail to find organisation (nice)" in {
+    val organisationId = SnykDisplay.getOrganisation(SnykDisplayTest.mockBadResponseWithMessage, SnykOrganisationName("guardian"))
+    organisationId.isFailedAttempt shouldBe true
+    organisationId.getFailedAttempt.failures.head.friendlyMessage shouldBe "Could not read Snyk response (some nice error)"
+  }
+
+  "fail to find organisation (not nice)" in {
+    val organisationId = SnykDisplay.getOrganisation(SnykDisplayTest.mockBadResponseWithoutMessage, SnykOrganisationName("guardian"))
+    organisationId.isFailedAttempt shouldBe true
+    organisationId.getFailedAttempt.failures.head.friendlyMessage shouldBe """Could not read Snyk response ({"toughluck": "no use"})"""
+  }
+
+  "find projects" - {
+
+    "find project 1 in project list" in {
+      val projects = SnykDisplay.getProjectIdList(SnykDisplayTest.mockGoodProjectResponse)
+      projects.value().exists(p => p.name == "project1") shouldBe true
+    }
+
+    "find project 2 in project list" in {
+      val projects = SnykDisplay.getProjectIdList(SnykDisplayTest.mockGoodProjectResponse)
+      projects.value().exists(p => p.name == "project2") shouldBe true
+    }
+
+  }
+
+  "fail to find project list" - {
```

Comment:
Should this not be part of the group above?

Reply:
(+general grouping in this file)

Reply:
Done